### PR TITLE
DropButtonProps TS improvement [3165]

### PR DESCRIPTION
**Pull request by fredhase**
#### What does this PR do?
Fix the types in src/js/components/DropButton/index.d.ts and add src/js/components/DropButton/stories/typescript/Calendar.tsx for validation.

#### Where should the reviewer start?
Both files mentioned above

#### What testing has been done on this PR?
Manual testing and the tests that already exists on the project

#### How should this be manually tested?
Testing on Storybook

#### What are the relevant issues?
#3165 

#### Do the grommet docs need to be updated?
I believe not.

#### Should this PR be mentioned in the release notes?
I believe not.

#### Is this change backwards compatible or is it a breaking change?
I believe it is.

### CODE REVIEW

File: src/js/components/DropButton/stories/typescript/Calendar.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        <DropButton
+          open={open}
+          onClose={() => setOpen(false)}
+          onOpen={() => setOpen(true)}
+          dropContent={<Calendar date={date} onSelect={onSelect} />}
+        >
+          <Box direction="row" gap="medium" align="center" pad="small">
+            <Text>
+              {date ? new Date(date).toLocaleDateString() : 'Select date'}
+            </Text>
```
Comment (negreirosleo) [inline src/js/components/DropButton/stories/typescript/Calendar.tsx]: I think you dont need this here, you could do:
```js
const [date, setDate] = useState('Select date') // initial state

const onSelect = (selectedDate) => {
    formattedDate = new Date(date).toLocaleDateString() //format date to localeDateString
    setDate(formattedDate);
    setOpen(false);
}

//...

<Text>{date}</Text> // no need for conditional
```

File: src/js/components/DropButton/stories/typescript/Calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { FormDown } from 'grommet-icons';
+
+import { Grommet, Box, Calendar, DropButton, Text } from 'grommet';
+import { grommet } from 'grommet/themes';
+
+const CalendarDropButton = () => {
+  const [date, setDate] = React.useState();
```
Comment (negreirosleo) [inline src/js/components/DropButton/stories/typescript/Calendar.tsx]: how about:
```js
import React, { useState } from 'react'

//...

  const [date, setDate] = useState()
  const [open, setOpen] = useState()

**Comment by ShimiSun** [comment]
Any updates on this PR @fredhase ?

**Comment by fredhase** [comment]
Sorry, @ShimiSun, I've had some tough weeks at work. I'll start looking into my PRs now

**Comment by ShimiSun** [comment]
Hi @fredhase any updates on this one?

File: src/js/components/DropButton/index.d.ts [hunk 3]
```diff
@@ -8,8 +8,8 @@ export interface DropButtonProps {
   dropContent: JSX.Element;
   dropTarget?: object;
   dropProps?: DropProps;
-  onClose?: ((...args: any[]) => any);
-  onOpen?: ((...args: any[]) => any);
+  onClose?: (() => void);
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/DropButton/index.d.ts:11]: Yes, an event needs to be accepted here. Applying the above code causes an error. For now, let's revert the parameter to args any, but we can leave the return as void.

File: src/js/components/DropButton/stories/typescript/Calendar.tsx (hunk 2 above)
Comment (taysea) [reply src/js/components/DropButton/stories/typescript/Calendar.tsx:9]: Incorporating this cleanup.

**Pull request closed without merging**